Lux/src/fimage.cpp

```diff
@@ -0,0 +1,69 @@
+// Floating point image class using image template as a base
+
+#include "fimage.hpp"
+#include <memory>
+#include "image_loader.hpp"
+
+// pixel modification functions
+void fimage :: grayscale() {
+    std :: transform( base.begin(), base.end(), base.begin(),  []( frgb &f ) { return gray( f ); } );
```

Comment:
can choose vscode option "format document" to automatically fix this inconsistent indentation

Reply:
I will examine this feature. I like to have spaces within the code as I think it makes it more readable. The most important thing is to do it consistently.